test/spec/SpecLanguageAutocomplete.js

```diff
@@ -52,4 +52,50 @@ describe('Test Language Selector Autocomplete', function () {
     });
   });
 
+  it('Adding and removing a button for "en"', function (done) {
+    expect(getLanguageCodesToFilter()).toEqual([]);
+
+    addLanguageFilter('en');
+    expect($('#lc-filter-en').length).toEqual(1);
+
+    // Adding again should not make another button
+    addLanguageFilter('en');
+    expect($('#lc-filter-en').length).toEqual(1);
+
+    // Adding "fr" should result in two buttons
+    addLanguageFilter('fr');
+    expect($('.lc-filter').length).toEqual(2);
+    expect(getLanguageCodesToFilter()).toEqual(['en', 'fr']);
```

Comment:
@richmahn - When stepping through the tests, It looks like getLanguageCodesToFilter() on line 68 is returning `lcs = ["undefined"]` which is blowing up the test.